internal/strings/strings_test.go

```diff
@@ -89,3 +92,50 @@ func TestRandomStringConcurrency(t *testing.T) {
 		go RandomString(10000)
 	}
 }
+
+func TestAsciiToLower(t *testing.T) {
+	word := "HELLO WORLD"
+	if want, have := "hello world", AsciiToLower(word); want != have {
+		t.Errorf("Unexpected lowercase word, want %q, have %q", want, have)
+	}
+}
+
+func BenchmarkAsciiVsUnicodeCaseString(b *testing.B) {
+	strs := []string{
+		"!!!!!!!!!!!!!!aaaaaaaaa please merge me!!!",
```

Comment:
Ah also it looks like this is almost the same test case as "lowercase with many characters"

While I appreciate the sentiment, optimizations have no meaning without proper benchmarks to justify them, hence this is taking effort to merge ;)